### PR TITLE
Fix mkInitTokenPolicy token burning check

### PR DESCRIPTION
**Pull request by kpinter-iohk**
Some miscellaneous minor cleanups are also included in this PR.

### Prereview checklist

- [ ] The [CHANGELOG.md](../blob/master/CHANGELOG.md) has been updated under the `# Unreleased` header using the appropriate sub-headings with links to the appropriate issues/PRs.
- [ ] All tests pass in CI
- [ ] PR was self-reviewed

### CODE REVIEW

File: onchain/src/TrustlessSidechain/InitToken.hs [hunk 1]
```diff
@@ -60,10 +60,8 @@ mkInitTokenPolicy _ BurnInitToken ctx
             (AssocMap.lookup cs mintedValue)
 
         -- Check that we only burn tokens
-        -- TODO replace with `all` (or `any`? why the `||`?)
         allBurned :: [Integer] -> Bool
-        allBurned [] = True
-        allBurned (x : xs) = x < 0 || allBurned xs
```
Comment (kpinter-iohk) [inline onchain/src/TrustlessSidechain/InitToken.hs:66]: This `||` was very likely meant to be an `&&` originally. With `[]` evaluating to `True` and the conditions being ORed together `allBurned` always evaluates to `True`, meaning that the `InitToken` minting policy could mint new tokens with `BurnInitToken` redeemer.

Comment (gilligan) [reply onchain/src/TrustlessSidechain/InitToken.hs:66]: Gosh that is pretty bad. And obviously we have no test for this? Could we *have* a test for this?

File: onchain/src/TrustlessSidechain/Versioning.hs [hunk 2]
```diff
@@ -425,12 +425,9 @@ mkVersionOracleValidator
...
       versionOutputAbsent =
-        -- TODO replace with all after Plutus version upgrade
-        null
-          [ ()
-          | txOut <- Unsafe.txInfoOutputs txInfo
-          , valueOf (Unsafe.decode $ Unsafe.txOutValue txOut) currencySymbol versionOracleTokenName > 0
-          ]
+        all
+          (\txOut -> valueOf (Unsafe.decode $ Unsafe.txOutValue txOut) currencySymbol versionOracleTokenName > 0)
+          (Unsafe.txInfoOutputs txInfo)
```
Comment (jankun4) [inline onchain/src/TrustlessSidechain/Versioning.hs:430]: Why `null` got changed to `all`? Doesn't it make this work in the completely opposite way? We wanted to make sure that NO versioning tokens go out of the transaction that burns them.

Comment (kpinter-iohk) [reply onchain/src/TrustlessSidechain/Versioning.hs:430]: Good catch! I forgot to negate the condition. (So `not (any p) = all (not p)`)

File: onchain/src/TrustlessSidechain/InitToken.hs (hunk 1 above)
Comment (kpinter-iohk) [reply onchain/src/TrustlessSidechain/InitToken.hs:66]: It should be possible to construct a transaction that tries to mint a token when it shouldn't. I would do that in a follow up PR though.

**Review by jankun4 — APPROVED** [review]

**Pull request merged**